src/status_im2/contexts/chat/bottom_sheet_composer/view.cljs

```diff
@@ -21,152 +21,164 @@
+                        chat-input
+                        keyboard-height
+                        (boolean (seq images))
+                        reply
+                        edit
+                        audio)
+    (effects/setup-selection props)
+    (utils/update-input props state input-text)
+    [:<>
+     [mentions/view props state animations max-height cursor-pos]
+     [gesture/gesture-detector
+      {:gesture (drag-gesture/drag-gesture props state animations dimensions keyboard-shown)}
+      [reanimated/view
+       {:style     (style/sheet-container insets state animations)
+        :on-layout #(handler/layout % state blur-height)}
+       [sub-view/bar]
+       [reply/view state]
+       [edit/view edit #(utils/cancel-edit-message state animations)]
+       [reanimated/touchable-opacity
+        {:active-opacity      1
+         :on-press            (when @(:input-ref props) #(.focus ^js @(:input-ref props)))
+         :style               (style/input-container (:height animations) max-height)
+         :accessibility-label :message-input-container}
+        [rn/selectable-text-input
+         {:ref       #(reset! (:selectable-input-ref props) %)
+          :menuItems @(:menu-items state)
+          :style     (style/input-view props state)}
+         [rn/text-input
+          {:ref                      #(reset! (:input-ref props) %)
+           :default-value            @(:text-value state)
+           :on-focus                 #(handler/focus props state animations dimensions)
+           :on-blur                  #(handler/blur state animations dimensions images reply)
+           :on-content-size-change   #(handler/content-size-change %
+                                                                   state
+                                                                   animations
+                                                                   dimensions
+                                                                   (or keyboard-shown edit))
+           :on-scroll                #(handler/scroll % props state animations dimensions)
+           :on-change-text           #(handler/change-text % props state)
+           :on-selection-change      #(handler/selection-change % props state)
+           :on-selection             #(selection/on-selection % props state)
+           :max-height               max-height
+           :max-font-size-multiplier 1
+           :multiline                true
+           :placeholder              (i18n/label :t/type-something)
+           :placeholder-text-color   (colors/theme-colors colors/neutral-40 colors/neutral-50)
+           :style                    (style/input-text)
+           :accessibility-label      :chat-message-input}]]
+        [gradients/view props state animations show-bottom-gradient?]]
+       [images/images-list]
+       [actions/view props state animations window-height insets
+        (boolean (seq images))]]]]))
+
+(defn sheet-parent
```

Comment:
sheet-parent is confusing, it doesn't look like parent component , so we could just have two separate function, init-props and init-state, and use them in bottom-sheet-composer component?

Reply:
Done